jbmc/src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1398,9 +1399,12 @@ exprt java_string_library_preprocesst::make_argument_for_format(
     }
     else if(name==ID_int || name==ID_float || name==ID_char || name==ID_boolean)
     {
-      exprt value=get_primitive_value_of_object(
+      optionalt<symbol_exprt> value = get_primitive_value_of_object(
         arg_i, name, loc, symbol_table, code_not_null);
-      code_not_null.add(code_assignt(field_expr, value), loc);
+      if(value.has_value())
+        code_not_null.add(code_assignt(field_expr, *value), loc);
+      else
+        code_not_null.add(code_assignt(field_expr, nil_exprt()), loc);
```

Comment:
Puh, I'm pretty sure this is broken and has actually been all along? @romainbrenguier maybe?

Reply:
The else case should never happen, it would be better to put an invariant here.
The nil_exprt case happen for name == ID_void, which is excluded by the if condition.